pkg/virtual/syncer/strategies/transformers.go

```diff
@@ -0,0 +1,268 @@
+			workloadClusterName, err := syncercontext.WorkloadClusterNameFrom(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			numberOfFinalizers := len(newSyncerViewObject.GetFinalizers())
+			removeFromSyncer := newSyncerViewObject.GetDeletionTimestamp() != nil && numberOfFinalizers == 0
+
+			if bytes, err := yaml.Marshal(newSyncerViewObject.UnstructuredContent()); err == nil {
+				klog.V(5).Infof("Before SyncerViewTransformer - New Syncer View Resource:\n%s", string(bytes))
+			}
+
+			newSyncerViewObjectJson, err := json.Marshal(newSyncerViewObject.UnstructuredContent())
+			if err != nil {
+				return nil, err
+			}
+
+			oldKCPViewObject, err := client.Get(ctx, newSyncerViewObject.GetName(), metav1.GetOptions{ResourceVersion: newSyncerViewObject.GetResourceVersion()})
+			if err != nil {
+				return nil, err
+			}
+
+			oldSyncerViews, err := GetSyncerViews(oldKCPViewObject)
+			if err != nil {
+				return nil, err
+			}
+
+			if removeFromSyncer {
+				newSyncerViewObject = nil
+			}
+
+			syncing := GetSyncing(oldKCPViewObject)
+
+			newKCPViewObject, err = syncStrategy.UpdateFromSyncTarget(workloadClusterName, newSyncerViewObject, oldKCPViewObject, oldSyncerViews, syncing)
+			if err != nil {
+				return nil, err
+			}
+
+			newKCPViewAnnotations := newKCPViewObject.GetAnnotations()
+			if newKCPViewAnnotations == nil {
+				newKCPViewAnnotations = make(map[string]string)
+			}
+
+			newKCPViewObjectJson, err := newKCPViewObject.MarshalJSON()
+			if err != nil {
+				return nil, err
+			}
+
+			if removeFromSyncer {
+				delete(oldSyncerViews, workloadClusterName)
+			} else {
+				oldSyncerViews[workloadClusterName] = *newSyncerViewObject
+			}
+
+			newSyncing := GetSyncing(newKCPViewObject)
+
+			// Update syncer view diff for all sync targets based on the new KCP View object
+			for syncTarget, syncerView := range oldSyncerViews {
+				if syncTargetSyncing := newSyncing[syncTarget]; !syncTargetSyncing.Active() {
+					delete(newKCPViewAnnotations, syncerViewAnnotationPrefix+syncTarget)
+					continue
+				}
+
+				var syncerViewObjectJson []byte
+				if syncTarget == workloadClusterName {
+					syncerViewObjectJson = newSyncerViewObjectJson
+				} else {
+					syncerViewObjectJson, err = syncerView.MarshalJSON()
+					if err != nil {
+						return nil, err
+					}
+				}
+				syncerViewDiff, err := jsonpatch.CreateMergePatch(newKCPViewObjectJson, syncerViewObjectJson)
+				if err != nil {
+					return nil, err
+				}
+				newKCPViewAnnotations[syncerViewAnnotationPrefix+syncTarget] = string(syncerViewDiff)
+			}
+			if removeFromSyncer {
+				delete(newKCPViewAnnotations, syncerViewAnnotationPrefix+workloadClusterName)
+			}
+			newKCPViewObject.SetAnnotations(newKCPViewAnnotations)
+
+			if bytes, err := yaml.Marshal(newKCPViewObject.UnstructuredContent()); err == nil {
+				klog.V(5).Infof("Before SyncerViewTransformer - New KCP View Resource:\n%s", string(bytes))
+			}
+
+			return newKCPViewObject, nil
+		},
```

Comment:
this functions need `// doing X` every few lines. It's far too long to be totally without structure.